tracer/src/Datadog.Trace/IAST/HashBasedDeduplication.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="HashBasedDeduplication.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Iast;
+
+internal static class HashBasedDeduplication
+{
+    public const int MaximumSize = 1000;
+    private static HashSet<int> vulnerabilityHashes;
+
+    static HashBasedDeduplication()
+    {
+        vulnerabilityHashes = new HashSet<int>();
+    }
+
+    public static void Clear()
+    {
+        lock (vulnerabilityHashes)
+        {
+            vulnerabilityHashes.Clear();
+        }
+    }
+
+    public static bool Add(Vulnerability vulnerability)
+    {
+        var hashCode = vulnerability.GetHashCode();
+
+        bool newVulnerability;
+        lock (vulnerabilityHashes)
+        {
+            newVulnerability = vulnerabilityHashes.Add(hashCode);
```

Comment:
Would it make sense to have a queue along side the set to remove only one item when it's full? Would it be worse feature wise irt to the extra memory?

Reply:
Yes, it makes a lot of sense. I have talked to @smola about it and he thinks that it is a good idea that the iast teams can discuss. The Iast teams still have pending to implement a better mechanism when the set is full, but a queue would be a nice option. Thank you for the idea!

I will include to @anderruiz

Reply:
Just to point out, even if you don't want the extra memory of a queue, you could use a `ConcurrentDictionary<int, bool>` instead of a `HashSet<int>` to avoid all the extra locks etc.

Reply:
> but a queue would be a nice option

In case you go for this change one day, I realized we already have this implementation of dedup and queue in the [RareSampler](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Agent/TraceSamplers/RareSampler.cs)  so there would be an opportunity to mutualize it.